controllers/mirrorpeer_controller.go

```diff
@@ -161,9 +193,92 @@ func processMirrorPeerSecretChanges(ctx context.Context, rc client.Client, mirro
 	return anyErr
 }
 
+func (r *MirrorPeerReconciler) checkTokenExchangeStatus(ctx context.Context, mp multiclusterv1alpha1.MirrorPeer) (bool, error) {
+	logger := log.FromContext(ctx)
+
+	//TODO Add support for more peer refs when applicable
+	pr1 := mp.Spec.Items[0]
+	pr2 := mp.Spec.Items[1]
+
+	err := r.checkForSourceSecret(ctx, pr1)
+	if err != nil {
+		logger.Error(err, "Failed to find valid source secret", "PeerRef", pr1)
+		return false, err
+	}
+
+	err = r.checkForDestinationSecret(ctx, pr1, pr2.ClusterName)
+	if err != nil {
+		logger.Error(err, "Failed to find valid destination secret", "PeerRef", pr1)
+		return false, err
+	}
+
+	err = r.checkForSourceSecret(ctx, pr2)
+	if err != nil {
+		logger.Error(err, "Failed to find valid source secret", "PeerRef", pr1)
+		return false, err
+	}
+
+	err = r.checkForDestinationSecret(ctx, pr2, pr1.ClusterName)
+	if err != nil {
+		logger.Error(err, "Failed to find destination secret", "PeerRef", pr2)
+		return false, err
+	}
+
+	return true, nil
+}
+
+func (r *MirrorPeerReconciler) checkForSourceSecret(ctx context.Context, peerRef multiclusterv1alpha1.PeerRef) error {
+	logger := log.FromContext(ctx)
+	prSecretName := common.CreateShortUniqueName(peerRef.ClusterName, peerRef.StorageClusterRef.Namespace, peerRef.StorageClusterRef.Name)
```

Comment:
`CreateShortUniqueName` takes n-params and here we are making an unwritten convention that the first argument should be clustername, second namespace and third a storagecluster name.
It would have been better if we have a particular function like `CreateUniqueSecretName` which takes exactly three strings with params specifically saying what to pass.

Reply:
In this way, if we change the params for uniqueness, we have to modify this general function and everyone will be aware of that extra parameter change.

Reply:
WIll follow up on this later

Reply:
should be fixed after rebasing on talur's PR